packages/pds/src/sequencer/sequencer-leader.ts

```diff
@@ -0,0 +1,127 @@
+import { DisconnectError } from '@atproto/xrpc-server'
+import { jitter, wait } from '@atproto/common'
+import { Leader } from '../db/leader'
+import { seqLogger as log } from '../logger'
+import Database from '../db'
+
+export const SEQUENCER_LEADER_ID = 1100
+
+export class SequencerLeader {
+  leader: Leader
+
+  destroyed = false
+  polling = false
+  queued = false
+  lastSeq: number
+
+  constructor(public db: Database) {
+    const lockId = this.db.getSchemaAdvisoryLockId(SEQUENCER_LEADER_ID)
+    this.leader = new Leader(lockId, this.db)
+  }
+
+  nextSeqVal(): number {
+    this.lastSeq++
+    return this.lastSeq
+  }
+
+  async run() {
+    if (this.db.dialect === 'sqlite') return
+
+    const res = await this.db.db
+      .selectFrom('repo_seq')
+      .select('seq')
+      .orderBy('seq', 'desc')
+      .limit(1)
+      .executeTakeFirst()
+    this.lastSeq = res?.seq ?? 0
```

Comment:
The `lastSeq` could change by the time this node has become leader, in which case we'd probably move this down into the `leader.run()` callback.

Reply:
I think there's also totally a way to push the `lastSeq` state down into a postgres sequence if that simplifies the bookkeeping in the leader.

Reply:
oh shit good call 👍